store/siren-action-behavior.js

```diff
@@ -103,7 +103,15 @@ D2L.PolymerBehaviors.Siren.SirenActionBehaviorImpl = {
 			.then(function(resp) {
 				if (!resp.ok) {
 					var errMsg = resp.statusText + ' response executing ' + opts.method + ' on ' + href + '.';
-					return resp.json().then(function(data) {
+					return resp.text()
+					.then(function(text) {
+						try {
+							return JSON.parse(text);
```

Comment:
we could probably just use `resp.json` in case it has extra magic sauce vs `JSON.parse`

Reply:
when I'm inside the async code, I immediately lose the `resp`. So I can't do this
```javascript
resp.json().catch(function(text) {
  return resp.text(); // resp is undefined :( 
```
the code that I do have access code, is whatever is in the arguments, hence my approach

Reply:
(caveat if I use `const` or `window.someVar` that should stay but I haven't tested it)

Reply:
that seems pretty suspicious to me, but maybe I'm missing something. It should be accessible inside the closure, unless something is explicitly setting it to `undefined`